packages/go/cypher/models/pgsql/translate/expansion.go

```diff
@@ -712,6 +696,111 @@ func (s *Translator) buildExpansionPatternRoot(part *PatternPart, traversalStep
 	return expansion.Build(traversalStep.Expansion.Value.Binding.Identifier), nil
 }
 
+type ExpansionComponents struct {
+	TerminalNode         *BoundIdentifier
+	RootNode             *BoundIdentifier
+	RightNodeConstraints *Constraint
+	EdgeConstraints      *Constraint
+	LeftNodeConstraints  *Constraint
+	PrimerVisible        *pgsql.IdentifierSet
+	RecursiveVisible     *pgsql.IdentifierSet
+	PrimerWhereClause    pgsql.Expression
+	RecursiveWhereClause pgsql.Expression
+}
+
+func PrepareExpansionRootComponents(part *PatternPart, traversalStep *PatternSegment, treeTranslator *ExpressionTreeTranslator) (ExpansionComponents, error) {
```

Comment:
Similar nit feedback here re: private vs public

Reply:
Patching this up.